back/src/Guide/Db/Get.hs

```diff
@@ -0,0 +1,305 @@
+
+-- | Get category by item uid.
+getCategoryByItem :: Uid Item -> Session (Maybe Category)
+getCategoryByItem itemId = do
+  catId <- getCategoryIdByItem itemId
+  getCategoryMaybe catId
+
+-- | Get category's uid list
+getCategoryIds :: Session [Uid Category]
+getCategoryIds = do
+  let sql = [r|
+        SELECT uid
+          FROM categories
+        ;|]
+      encoder = HE.noParams
+      decoder = HD.rowList $ Uid <$> textDecoderNonNull
+  HS.statement () (Statement sql encoder decoder False)
+
+-- | Get all categories
+getCategories :: Session [Category]
+getCategories = do
+  catIds <- getCategoryIds
+  traverse getCategory catIds
+
+--
+-- | General functions
+--
+
+--
+-- | Encoders
+--
+
+-- | Encoder parameter for text
+textEncoderNonNull :: HE.Params Text
+textEncoderNonNull = HE.param (HE.nonNullable HE.text)
+
+-- | Encoder parameter for text
+traitTypeEncoderNonNull :: HE.Params TraitType
+traitTypeEncoderNonNull = HE.param (HE.nonNullable $ HE.enum (T.pack . show))
+
+-- | Encoder parameter for bool
+boolEncoderNonNull :: HE.Params Bool
+boolEncoderNonNull = HE.param (HE.nonNullable HE.bool)
+
+--
+-- | Decoders
+--
+
+-- | Decoder for nonNull text
+textDecoderNonNull :: HD.Row Text
+textDecoderNonNull = (HD.column . HD.nonNullable) HD.text
+
+-- | Decoder row for maybe text
+textDecoderNull :: HD.Row (Maybe Text)
+textDecoderNull = (HD.column . HD.nullable) HD.text
+
+-- | Decoder for [text]
+textArrayDecoder :: HD.Row [Text]
+textArrayDecoder = (HD.column . HD.nonNullable)
+  $ HD.array (HD.dimension replicateM (HD.element $ HD.nonNullable HD.text))
+
+--
+-- | Rows
+--
+
+-- | Trait row
+traitRow :: HD.Row Trait
+traitRow = do
+  _traitUid <- Uid <$> textDecoderNonNull
+  _traitContent <- toMarkdownInline <$> textDecoderNonNull
+  pure $ Trait{..}
+
+-- | Item row
+itemRow :: Text -> [Trait] -> [Trait] -> [Trait] -> [Trait] -> HD.Row Item
+itemRow pref _itemPros _itemProsDeleted _itemCons _itemConsDeleted = do
+  _itemUid <- Uid <$> textDecoderNonNull
+  _itemName <- textDecoderNonNull
+  _itemCreated <- localTimeToUTC utc <$> (HD.column . HD.nonNullable) HD.timestamp
+  _itemGroup_ <- textDecoderNull
+  _itemLink <- textDecoderNull
+  _itemHackage <- textDecoderNull
+  _itemSummary <- toMarkdownBlock <$> textDecoderNonNull
+  _itemEcosystem <- toMarkdownBlock <$> textDecoderNonNull
+  _itemNotes <- toMarkdownTree pref <$> textDecoderNonNull
+  pure $ Item{..}
+
+-- | Category row
+categoryRow :: [Item] -> [Item] -> HD.Row Category
+categoryRow _categoryItems _categoryItemsDeleted = do
+  _categoryUid <- Uid <$> textDecoderNonNull
+  _categoryTitle <- textDecoderNonNull
+  _categoryCreated <- localTimeToUTC utc <$> (HD.column . HD.nonNullable) HD.timestamp
+  _categoryGroup_ <- textDecoderNonNull
+  _categoryStatus <- toStatus <$> textDecoderNonNull
+  _categoryNotes <- toMarkdownBlock <$> textDecoderNonNull
+  selectionsText <- textArrayDecoder
+  let _categoryEnabledSections = Set.fromList $ map toSection selectionsText
```

Comment:
I don't like `toSection` and `toStatus` – they kinda depend on our internal type names. I'd rather have conversions inside `Guide.Database`. While I'm at it, we can have just "stub", "wip", "finished", there's no need for CategoryStatusBlahBlahStub.